eng/Versions.props

```diff
@@ -39,7 +39,7 @@
          repository. This must be lower than MicrosoftNetCompilersToolsetVersion,
          but not higher than our minimum dogfoodable Visual Studio version, or else
          the generators we build would load on the command line but not load in IDEs. -->
-    <SourceGeneratorMicrosoftCodeAnalysisVersion>3.8.0-3.final</SourceGeneratorMicrosoftCodeAnalysisVersion>
```

Comment:
This should also come with an update to our contributing docs that we now require 16.8 non-preview, lest we violate the constraint in the comment. (This should be easy as just deleting the "preview" bits in the markdown file.)

Reply:
done